content/reference/sql/version-control/dolt-sql-functions.md

```diff
@@ -798,3 +799,170 @@ With result of single row:
+for the defined table only.
+
+### Options
+
+```sql
+DOLT_SCHEMA_DIFF(<from_commit>, <to_commit>, <optional_tablename>)
+DOLT_SCHEMA_DIFF(<from_revision..to_revision>, <optional_tablename>)
+DOLT_SCHEMA_DIFF(<from_revision...to_revision>, <optional_tablename>)
+```
+
+The `DOLT_SCHEMA_DIFF()` table function takes three arguments:
+
+- `from_revision` — the revision of the table data for the start of the diff. This argument is required. This may be a commit, tag, branch name, or other revision specifier (e.g. "main~", "WORKING", "STAGED").
+- `to_revision` — the revision of the table data for the end of the diff. This argument is required. This may be a commit, tag, branch name, or other revision specifier (e.g. "main~", "WORKING", "STAGED").
+- `from_revision..to_revision` — gets the two dot diff, or revision of table schema between the `from_revision` and `to_revision`. This is equivalent to `dolt_schema_diff(<from_revision>, <to_revision>, <tablename>)`.
+- `from_revision...to_revision` — gets the three dot diff, or revision of table schema between the `from_revision` and `to_revision`, _starting at the last common commit_.
+- `tablename` — the name of the table to diff. This argument is optional. When it's not defined, all tables with schema diffs will be returned.
+
+### Schema
+
+```text
++-----------------------+------+
+| field                 | type |
++-----------------------+------+
+| from_table_name       | TEXT |
+| to_table_name         | TEXT |
+| from_create_statement | TEXT |
+| to_create_statement   | TEXT |
+| alter_statement       | TEXT |
++-----------------------+------+
+```
+
+### Example
+
+Consider a table named `inventory` in a database with two branches: `main` and `feature_branch`. We can use the
+`DOLT_SCHEMA_DIFF()` function to calculate a diff of the table schema from the `main` branch to the `feature_branch` branch
+to see how the table's schema has changed on the feature branch.
+
+Here is the schema of `inventory` at the tip of `main`:
+
+```text
++----------+------+
+| field    | type |
++----------+------+
+| pk       | int  |
+| name     | text |
+| quantity | int  |
++----------+------+
+```
+
+Here is the schema of `inventory` at the tip of `feature_branch`:
+
+```text
++----------+------+
+| field    | type |
++----------+------+
+| pk       | int  |
+| name     | text |
+| color    | text |
++----------+------+
+```
+
+To calculate the schema diff and view the results, we run the following query:
+
+```sql
+SELECT * FROM DOLT_SCHEMA_DIFF("main", "feature_branch", "inventory")
+```
+
+The results from `DOLT_SCHEMA_DIFF()` show how the schema has changed going from tip of `main` to tip of `feature_branch`:
+
+```text
++-----------------+---------------+-------------------------------------------------------------------+-------------------------------------------------------------------+--------------------------------------------------+
```

Comment:
(minor) Not strictly required, but might be fun/cool/nice to try out the new embedded DoltHub Workbench UI that @liuliu-dev just added. There are some examples in the system tables page. It's cool because users get the read the docs and actually run queries and see results while they're learning about the functions. Cool to follow up with that separately if you need to stay focused on other CLI migration work though.

Reply:
Working with Liu Liu on getting some schema changes into one of our example dolthub DBs, then will embed a query here.

Reply:
sweet!

Reply:
PR updated with an embed link with a DoltHub SQL query.
It *seems* to be comparable to other embed links, but I'm not sure. Is there a way to build docs locally?

Reply:
I think the only way to see it will be merging it to `gitbook-dev` and then checking it on dev: https://dolt.gitbook.io/dolt-dev/sql-reference/version-control/dolt-system-tables
It might take a few minutes to sync to dev after merging the PR.